lib/fluent/plugin/out_irc.rb

```diff
@@ -116,11 +124,25 @@ def emit(tag, es, chain)
       end
 
       es.each do |time,record|
+        if @send_queue.size >= @send_queue_limit
+          log.warn "out_irc: send queue size exceeded send_queue_limit(#{@send_queue_limit}), discards"
+          break
+        end
+
         filter_record(tag, time, record)
-        @conn.send_message(build_message(record), build_channel(record), build_command(record))
+        command, channel, message = build_command(record), build_channel(record), build_message(record)
+        log.debug { "out_irc: push {command:\"#{command}\", channel:\"#{channel}\", message:\"#{message}\"}" }
+        @send_queue.push([command, channel, message])
       end
     end
 
+    def on_timer
+      return if @send_queue.empty?
+      command, channel, message = @send_queue.shift
+      log.info { "out_irc: send {command:\"#{command}\", channel:\"#{channel}\", message:\"#{message}\"}" }
```

Comment:
OK. As this is trivial,  I will merge this PR and release it at first. I would try to solve the problem if I were annoyed with log messages. Thank you!